src/app/administrator/override/override.component.ts

```diff
@@ -30,24 +30,20 @@ export class OverrideComponent implements OnInit {
         },
         (error) => {
           this.overridableLoans = [];
-          // do something with a logger here
-          alert(error);
+          alert(error.error);
         }
       );
   }
 
   doOverride(loan: any) {
+    if (!confirm("Override the due date of this loan?")) return;
     const dateOut = moment(loan.dateOut).format("YYYY-MM-DDTHH_mm_ss"),
       overrideUri = `/loans/book/${loan.bookId}/borrower/${loan.cardNo}/branch/${loan.branchId}/dateout/${dateOut}`;
     this.lmsService
       .put(`${environment.adminBackendUrl}${overrideUri}`)
-      .subscribe(
-        () => this.readOverridableLoans(),
-        (error) => {
-          // do something with a logger here
-          alert(error);
-          this.readOverridableLoans();
-        }
-      );
+      .subscribe(null, (error) => {
```

Comment:
What is this null handling?

Reply:
That's just passing null as the function to execute if there's no error. I put reloading the data in add() so it gets done whether there's an error or not.